packages/components/src/internal/components/search/FilterFacetedSelector.tsx

```diff
@@ -1,64 +1,162 @@
+
+    const checkedValues = useMemo(() => {
+        return getCheckedFilterValues(fieldFilter, fieldDistinctValues);
+    }, [fieldFilter, fieldKey, fieldDistinctValues]); // need to add fieldKey
+
+    const taggedValues = useMemo(() => {
+        if (checkedValues?.indexOf(ALL_VALUE_DISPLAY) > -1) return [];
+        return checkedValues;
+    }, [checkedValues]);
+
+    const onSearchStrChange = useCallback(e => {
+        setSearchStr(e.target.value);
+    }, []);
+
+    const onChange = useCallback(
+        (value: string, checked: boolean, uncheckOthers?: boolean) => {
+            const newFilter = getUpdatedChooseValuesFilter(
+                fieldDistinctValues,
+                fieldKey,
+                value,
+                checked,
+                fieldFilter,
+                uncheckOthers
+            );
+            onFieldFilterUpdate(newFilter);
+        },
+        [fieldDistinctValues, fieldKey, fieldFilter, onFieldFilterUpdate]
+    );
+
+    const filteredFieldDistinctValues = useMemo(() => {
+        if (!searchStr) return fieldDistinctValues;
+
+        return fieldDistinctValues?.filter(val => {
+            let valuesToKeep = [];
+            if (checkedValues?.indexOf(ALL_VALUE_DISPLAY) === -1) valuesToKeep = checkedValues;
+            return valuesToKeep?.indexOf(val) > -1 || val?.toLowerCase().indexOf(searchStr.toLowerCase()) > -1;
         });
-    }, [activeFieldDistinctValues, selectDistinctOptions]);
+    }, [fieldDistinctValues, searchStr]);
 
     return (
         <>
             {error && <Alert>{error}</Alert>}
-            {!activeFieldDistinctValues && <LoadingSpinner />}
-            <div className="list-group search-parent-entity-col">
-                <ul className="nav nav-stacked labkey-wizard-pills">
-                    {activeFieldDistinctValues?.map((value, index) => {
-                        let displayValue = value;
-                        if (value === null || value === undefined) displayValue = '[blank]';
-                        if (value === true) displayValue = 'TRUE';
-                        if (value === false) displayValue = 'FALSE';
-                        return (
-                            <li key={index}>
-                                <div className="form-check">
-                                    <input
-                                        className="form-check-input"
-                                        type="checkbox"
-                                        name={'field-value-' + index}
-                                        disabled={true}
-                                        checked={false}
-                                    />
-                                    <span style={{ marginLeft: 5 }}>{displayValue}</span>
-                                </div>
-                            </li>
-                        );
-                    })}
-                </ul>
+            {!fieldDistinctValues && <LoadingSpinner />}
```

Comment:
if there was an error loading the distinct values, would the loading spinner still show here? should the `error` be added to the conditional here or maybe `setFieldDistinctValues` should be called in the error case to set it to an empty array/list?